jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -654,7 +677,158 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
                 self.subset_definitions[index][i]['value'] = new_value
                 break
 
-    def _import_spectral_regions(self, spec_region, mode=NewMode, viewer=None):
+    def import_region(self, region, **kwargs):
+        return_bad_regions = kwargs.pop('return_bad_regions', None)
+        max_num_regions = kwargs.pop('max_num_regions', None)
+        refdata_label = kwargs.pop('refdata_label', None)
+
+        if isinstance(region, str):
+            if os.path.exists(region):
+                from regions import Regions
+                region_format = kwargs.pop('region_format', None)
+                try:
+                    raw_regs = Regions.read(region, format=region_format)
+                except:  # noqa
+                    raw_regs = SpectralRegion.read(region)
+
+                return self._load_regions(raw_regs, max_num_regions, refdata_label,
+                                          return_bad_regions, **kwargs)
+        else:
+            return self._load_regions(region, max_num_regions, refdata_label,
+                                      return_bad_regions, **kwargs)
+
+    def _load_regions(self, regions, max_num_regions=None, refdata_label=None,
+                      return_bad_regions=False, **kwargs):
+        if len(self.app.data_collection) == 0:
+            raise ValueError('Cannot load regions without data.')
+
+        from photutils.aperture import (CircularAperture, SkyCircularAperture,
+                                        EllipticalAperture, SkyEllipticalAperture,
+                                        RectangularAperture, SkyRectangularAperture,
+                                        CircularAnnulus, SkyCircularAnnulus)
+        from regions import (Regions, CirclePixelRegion, CircleSkyRegion,
+                             EllipsePixelRegion, EllipseSkyRegion,
+                             RectanglePixelRegion, RectangleSkyRegion,
+                             CircleAnnulusPixelRegion, CircleAnnulusSkyRegion)
+        from jdaviz.core.region_translators import regions2roi, aperture2regions
+
+        # If user passes in one region obj instead of list, try to be smart.
+        if (isinstance(regions, SpectralRegion) or
+                (isinstance(regions, list) and isinstance(regions[0], SpectralRegion))):
+            mode = kwargs.pop('mode', None)
+            self._import_spectral_regions(regions, mode)
+            return
```

Comment:
1. Given that you `return` here, all the spatial region imports (if you still want them locally scoped) should happen after this.
2. You should `return bad_regions` here to be consistent with spatial loader behavior, which means your `_import_spectral_regions` should give you back `bad_regions` like the spatial one does.

```suggestion
            return self._import_spectral_regions(regions, mode)
```